modifications/DataClass.cc

```diff
@@ -86,29 +97,34 @@ void DataClass::checkDroneStatus(QString location)
     request.setRawHeader("Authorization",QByteArray("Bearer ").append(accessToken));
     QNetworkReply* reply = manager.get(request);
     connect(reply,&QNetworkReply::finished, this, &DataClass::readyReadDroneStatus);
-
 }
 
 void DataClass::readyReadDroneStatus()
 {
     QNetworkReply* reply = qobject_cast<QNetworkReply*>(sender());
     QString replyStr = reply->readAll();
-    QJsonDocument jsonResponse = QJsonDocument::fromJson(replyStr.toUtf8());
-    QJsonObject jsonObject = jsonResponse.object();
-    drone.status = jsonObject["status"].toInt();
-    drone.uuid = jsonObject["id"].toString();
-    if(drone.status){
-        emit droneActive();
-    }else{
-        emit droneNotActive();
+    int statusCode = reply->attribute( QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    if(statusCode == 201 || statusCode == 200){
+        QJsonDocument jsonResponse = QJsonDocument::fromJson(replyStr.toUtf8());
+        QJsonObject jsonObject = jsonResponse.object();
+        drone.status = jsonObject["status"].toInt();
+        drone.uuid = jsonObject["id"].toString();
+        if(drone.status){
+            emit droneActive();
+        }else{
+            emit droneNotActive();
+            showMessage("Drone Not Active", -1);
+        }
+    }else if(statusCode == 404){
```

Comment:
Again, 404 should be as a part of the status enum/defines. Need not have it for all status codes but the most used ones.